cub/test/catch2_test_device_histogram.cu

```diff
@@ -0,0 +1,662 @@
+
+  const auto num_levels  = array<int, ActiveChannels>{11, 3, 2};
+  const auto lower_level = array<int, ActiveChannels>{0, -10, std::numeric_limits<int>::lowest()};
+  const auto upper_level = array<int, ActiveChannels>{total_values, 10, std::numeric_limits<int>::max()};
+
+  auto sample_iterator = cub::CountingInputIterator<sample_t>(0);
+
+  // Channel #0: 0, 4,  8, 12
+  // Channel #1: 1, 5,  9, 13
+  // Channel #2: 2, 6, 10, 14
+  // unused:     3, 7, 11, 15
+
+  auto d_histogram = array<c2h::device_vector<int>, ActiveChannels>();
+  for (int c = 0; c < ActiveChannels; ++c)
+  {
+    d_histogram[c].resize(num_levels[c] - 1);
+  }
+
+  multi_histogram_even<Channels, ActiveChannels>(
+    sample_iterator,
+    to_ptr_array(d_histogram).data(),
+    num_levels.data(),
+    lower_level.data(),
+    upper_level.data(),
+    width,
+    height,
+    row_pitch);
+
+  // TODO(bgruber): move detail::to_vec into namespace c2h?
+  CHECK(detail::to_vec(d_histogram[0]) == std::vector<int>(10, (width * height) / 10));
+  CHECK(detail::to_vec(d_histogram[1]) == std::vector<int>{0, 3});
+  CHECK(detail::to_vec(d_histogram[2]) == std::vector<int>{width * height});
+}
+
+// Regression: https://github.com/NVIDIA/cub/issues/479
+CUB_TEST("DeviceHistogram::Histogram* regression NVIDIA/cub#479", "[histogram][device]")
+{
+  test_even_and_range<float, 4, 3, int>(12, 7, 1920, 1080);
+}
+
+CUB_TEST("DeviceHistogram::Histogram* down-conversion size_t to int", "[histogram][device]")
+{
+  _CCCL_IF_CONSTEXPR (sizeof(std::size_t) != sizeof(int))
+  {
+    using offset_t = std::make_signed<std::size_t>::type;
+    test_even_and_range<unsigned char, 4, 3, int>(256, 256 + 1, offset_t{1920}, offset_t{1080});
+  }
+}
+
+CUB_TEST("DeviceHistogram::HistogramRange levels/samples aliasing", "[histogram_range][device]")
+{
+  constexpr int num_levels = 7;
+  constexpr int h_samples[]{
+    0,  2,  4,  6,  8,  10, 12, // levels
+    1, // bin 0
+    3,  3, // bin 1
+    5,  5,  5, // bin 2
+    7,  7,  7,  7, // bin 3
+    9,  9,  9,  9,  9, // bin 4
+    11, 11, 11, 11, 11, 11 // bin 5
+  };
+
+  auto d_histogram = c2h::device_vector<int>(num_levels - 1);
+  auto d_samples   = c2h::device_vector<int>(std::begin(h_samples), std::end(h_samples));
+  histogram_range(
+    thrust::raw_pointer_cast(d_samples.data()),
+    thrust::raw_pointer_cast(d_histogram.data()),
+    num_levels,
+    thrust::raw_pointer_cast(d_samples.data()), // Alias levels with samples (fancy way to `d_histogram[bin]++`).
+    static_cast<int>(d_samples.size()));
+
+  auto h_histogram = c2h::host_vector<int>(d_histogram);
+  for (int bin = 0; bin < num_levels - 1; bin++)
+  {
+    // Each bin should contain `bin + 1` samples, plus one extra, since samples also contain levels.
+    CHECK(h_histogram[bin] == bin + 2);
+  }
+}
+
+// TODO(bgruber): this test checks error codes, is it fine if we check this only for host-side invocation?
+#if TEST_LAUNCH == 0
```

Comment:
I'd appreciate an opinion here. Thist test cannot use launch wrappers.

Reply:
That's fine. Before this PR, we haven't tested device launch at all before, so this is not a regression. Besides that, majorify of the lid_1 issues are caught at compile time. Code paths corresponding to overflow detection are going to be compiled, so we are good.